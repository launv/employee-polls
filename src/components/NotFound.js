import { useNavigate } from "react-router-dom";
import { Button } from "primereact/button";

const NotFound = () => {
  const navigate = useNavigate();

  return (
    <div className="text-center">
      <p className="my-40 text-4xl">Sorry, this page doesn't exist!</p>
      <Button onClick={() => navigate("/")} label="Back to Homepage"></Button>
    </div>
  );
};

export default NotFound;
