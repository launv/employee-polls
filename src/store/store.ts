import { createStore } from "redux";
import middleware from "../middleware/index";
import reducer from "../reducers/index";

const store = createStore(reducer, middleware);

export default store;
