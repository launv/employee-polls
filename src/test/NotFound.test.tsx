import * as React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../store/store";
import NotFound from "../components/NotFound";

describe("NotFound", () => {
  test("should match snapshot", async () => {
    const html = render(
      <MemoryRouter>
        <Provider store={store}>
          <NotFound />
        </Provider>
      </MemoryRouter>
    );

    expect(html).toMatchSnapshot();
  });
});
