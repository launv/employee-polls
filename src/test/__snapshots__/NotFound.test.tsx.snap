// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto text-center md:w-2/3"
      >
        <p
          class="my-40 text-4xl"
        >
          Sorry, this page doesn't exist!
        </p>
        <button
          aria-label="Back to Homepage"
          class="p-button p-component"
          data-pc-name="button"
          data-pc-section="root"
        >
          <span
            class="p-button-label p-c"
            data-pc-section="label"
          >
            Back to Homepage
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto text-center md:w-2/3"
    >
      <p
        class="my-40 text-4xl"
      >
        Sorry, this page doesn't exist!
      </p>
      <button
        aria-label="Back to Homepage"
        class="p-button p-component"
        data-pc-name="button"
        data-pc-section="root"
      >
        <span
          class="p-button-label p-c"
          data-pc-section="label"
        >
          Back to Homepage
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
